e2e/stack.go

```diff
@@ -53,89 +51,139 @@ func New(
 	}
 }
 
+// helper shim to allow decoding of hex nonces
+type auxDumpAccount struct {
+	Balance     string                 `json:"balance"`
+	Nonce       string                 `json:"nonce"`
+	Root        hexutil.Bytes          `json:"root"`
+	CodeHash    hexutil.Bytes          `json:"codeHash"`
+	Code        hexutil.Bytes          `json:"code,omitempty"`
+	Storage     map[common.Hash]string `json:"storage,omitempty"`
+	Address     *common.Address        `json:"address,omitempty"` // Address only present in iterative (line-by-line) mode
+	AddressHash hexutil.Bytes          `json:"key,omitempty"`     // If we don't have address, we can output the key
+}
+
+// helper shim to allow decoding of hex nonces
+type auxDump struct {
+	Root     string                    `json:"root"`
+	Accounts map[string]auxDumpAccount `json:"accounts"`
+	// Next can be set to represent that this dump is only partial, and Next
+	// is where an iterator should be positioned in order to continue the dump.
+	Next []byte `json:"next,omitempty"` // nil if no more accounts
+}
+
+func (d *auxDump) ToStateDump() (*state.Dump, error) {
+	accounts := make(map[string]state.DumpAccount)
+
+	for k, v := range d.Accounts { //nolint:gocritic
+		nonce, err := hexutil.DecodeUint64(v.Nonce)
+		if err != nil {
+			return nil, fmt.Errorf("decode nonce: %v", err)
+		}
+
+		accounts[k] = state.DumpAccount{
+			Balance:  v.Balance,
+			Nonce:    nonce,
+			Root:     v.Root,
+			CodeHash: v.CodeHash,
+			Code:     v.Code,
+			Storage:  v.Storage,
+		}
+	}
+
+	return &state.Dump{
+		Root:     d.Root,
+		Accounts: accounts,
+	}, nil
+}
+
 func (s *Stack) Run(ctx context.Context, env *environment.Env) error {
-	// Run anvil.
-	anvilCmd := exec.CommandContext( //nolint:gosec
-		ctx,
-		"anvil",
-		"--port", s.anvilURL.Port(),
-		"--order", "fifo",
-		"--disable-block-gas-limit",
-		"--gas-price", "0",
-		"--block-time", fmt.Sprint(s.l1BlockTime.Seconds()),
-	)
-	anvilCmd.Cancel = func() error {
-		// Anvil can catch SIGTERMs. The exec package sends a SIGKILL by default.
-		return anvilCmd.Process.Signal(syscall.SIGTERM)
+	// configure & run L1
+
+	const networkName = "devnetL1"
+	l1Deployments, err := opgenesis.NewL1Deployments(filepath.Join(s.l1stateDumpDir, "addresses.json"))
+	if err != nil {
+		return fmt.Errorf("new l1 deployments: %v", err)
 	}
-	if err := s.startCmd(anvilCmd); err != nil {
-		return err
+	deployConfig, err := opgenesis.NewDeployConfigWithNetwork(networkName, s.deployConfigDir)
+	if err != nil {
+		return fmt.Errorf("new deploy config: %v", err)
 	}
-	env.Go(func() {
-		if err := anvilCmd.Wait(); err != nil && !errors.Is(err, ctx.Err()) {
-			s.eventListener.OnAnvilErr(fmt.Errorf("run %s: %v", anvilCmd, err))
+	deployConfig.SetDeployments(l1Deployments)
+	deployConfig.L1UseClique = false // Allows node to produce blocks without addition config. Clique is a PoA config.
+
+	// Generate a deployer key and pre-fund the account
+	deployerKey, err := crypto.GenerateKey()
```

Comment:
Where is this account being pre-funded? Is that included in the state dump?

Reply:
This account isn't funded, which possibly will cause some downstream errors. It's also possible that this account is no longer needed - it had previously been the EOA that executed the L1 contract deployments via `forge` against the `anvil` backend. Not 100% clear to me that it had other responsibilities.

I have some work on prefunding accounts in the follow-up PR, so I'm happy enough to defer figuring this out for the moment.